platform/platform-configuration-listener/src/main/java/org/codice/ddf/configuration/store/ConfigurationFile.java

```diff
@@ -56,4 +62,49 @@ public Path getConfigFilePath() {
     }
 
     public abstract void createConfig() throws ConfigurationFileException;
+
+    public void exportConfig(String destination) throws IOException {
+        persistenceStrategy.write(new FileOutputStream(destination), properties);
+    }
+
+    /**
+     * Provides a convenient way to construct {@link ConfigurationFile}.
+     *
+     * @param configAdmin         reference to OSGi's {@link ConfigurationAdmin}
+     * @param persistenceStrategy how to write out the file {@link PersistenceStrategy}
+     */
+    public abstract static class ConfigurationFileBuilder {
```

Comment:
Since this is only used by the `ConfigurationFileFactory` class which is in the same package, I would either make this package-private or `protected`, not `public`.